mobile/lib/modules/home/ui/asset_grid/asset_grid_data_structure.dart

```diff
@@ -56,6 +56,24 @@ class RenderList {
   RenderList(this.elements, this.query, this.allAssets)
       : totalAssets = allAssets?.length ?? query!.countSync();
 
+  /// Creates a new render list from assets
+  factory RenderList.fromAssetsOnly(List<Asset> assets) {
+    // Guard empty assets
+    if (assets.isEmpty) {
+      return RenderList([], null, []);
+    }
+
+    final elements = assets
+        .map(
+          (a) => RenderAssetGridElement(
+            RenderAssetGridElementType.assets,
+            date: a.fileCreatedAt,
+          ),
+        )
+        .toList();
+    return RenderList(elements, null, assets);
+  }
+
```

Comment:
This is only used in one place (map_page) which I didn't test...

Reply:
We already got the static method fromAssets. Is that not enough?